hdf5/src/hl/references/legacy.rs

```diff
@@ -0,0 +1,69 @@
+//! The pre v1.12.0 reference types.
+
+use crate::internal_prelude::*;
+
+use hdf5_sys::{
+    h5o::H5O_type_t,
+    h5p::H5P_DEFAULT,
+    h5r::{hobj_ref_t, H5Rcreate, H5Rdereference, H5Rget_obj_type2},
+};
+use hdf5_types::H5Type;
+
+#[cfg(not(feature = "1.12.0"))]
+use hdf5_sys::h5r::H5R_OBJECT as H5R_OBJECT1;
+#[cfg(feature = "1.12.0")]
+use hdf5_sys::h5r::H5R_OBJECT1;
+
+use crate::{Location, ObjectReference};
+
+#[repr(transparent)]
+#[derive(Debug, Copy, Clone)]
```

Comment:
I don't think you can implement `Copy` and `Clone` for these?

Reply:
I was a little unclear on this. I went to remove it but these references aren't actually supported by H5Rdestroy or H5Rcopy. I'm not sure how to test for this in the absence of explicit documentation. I'll take a look at the HDF5 source for any clues

Reply:
Could be helpful to have a look in the header files of a 1.8 release

Reply:
Ah perhaps these should use H5O_close - I'll give this a go

Reply:
I've done a bit of digging and I think these can be copy but let me know if you think this is wrong:

* Found this example which shows them freed as normal memory: https://github.com/HDFGroup/hdf5/blob/1d1c0405449b70656d9ec30afa95d358fc2719a0/examples/h5_reference_deprec.c#L82
* The 1.8 docs show that it `hobj_ref_t` is an typdef for `haddr_t` which is a 64 bit address in the file.